src/wallet/wallet.cpp

```diff
@@ -2509,9 +2509,10 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, const CoinEligibil
     // Calculate the fees for things that aren't inputs
     CAmount not_input_fees = coin_selection_params.effective_fee.GetFee(coin_selection_params.tx_noinputs_size);
 
-    if (coin_selection_params.use_bnb) {
+    // Start with BnB. If that fails, use SRD
+    if (SelectCoinsBnB(utxo_pool, nTargetValue, cost_of_change, setCoinsRet, nValueRet, not_input_fees)) {
```

Comment:
I think this new logic means that non-BnB will be tried more often? Instead of trying all variants of BnB(6 confirms, 1 confirm, small chain, etc), we seem to be trying 6 confirms for BnB, then 6 confirms for non-BnB

Reply:
Hmm, yes, that appears to be what is happening. I guess this depends on whether we want to select an exact match for coins with less confirmatoins, or whether we want confirmations over exact match.

Reply:
This is a good point. Perhaps a solution is to add a `use_bnb` argument to `SelectCoinsMinConf` (but not as part `CoinSelectionParams`), and then have a sequence of attempts with and without use_bnb in `SelectCoins`?

Reply:
I prefer the behavior in master due to privacy reasons of change-less transactions.

Reply:
I have implemented this change.